dbt/include/global_project/macros/materializations/table/table.sql

```diff
@@ -48,8 +59,8 @@
   {% if non_destructive_mode -%}
     -- noop
   {%- else -%}
-    {{ drop_if_exists(existing, schema, identifier) }}
-    {{ adapter.rename(schema, tmp_identifier, identifier) }}
+    {{ drop_relation_if_exists(target_relation) }}
```

Comment:
I think this should use `old_relation` -- if it previously existed as a view, this will run `drop table ...` which will result in an error. There's code above (line 25) to drop said view, but that only runs in non destructive mode.